src/components/profile/EnvironmentForm.jsx

```diff
@@ -0,0 +1,218 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { withStyles } from "@material-ui/core";
+import { compose } from 'redux';
+import { connect } from "react-redux";
+import { getEnvironments } from "../../actions/environments";
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Dialog from '@material-ui/core/Dialog';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import api from '../../utils/api';
+import AddIcon from '@material-ui/icons/Add';
+import IconButton from '@material-ui/core/IconButton';
+import Input from '@material-ui/core/Input';
+import InputLabel from '@material-ui/core/InputLabel';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import FormControl from '@material-ui/core/FormControl';
+import Chip from '@material-ui/core/Chip';
+import List from '@material-ui/core/List';
+import Typography from "@material-ui/core/Typography";
+
+
+const styles = theme => ({
+	root:{
+		flex: 1,
+	},
+	container: {
+		display: 'flex',
+		flexWrap: 'wrap',
+	},
+	textField: {
+		margin: theme.spacing.unit*2,
+	},
+	tagStyle: {
+		margin: theme.spacing.unit,
+	},
+})
+
+class EnvironmentForm extends Component {
+  constructor(props) {
+		super(props);
+		this.state = {
+			id: this.props.envExists ? this.props.environment.id : this.props.repository.id,
+			name: this.props.envExists ? this.props.environment.name : this.props.repository.name,
+			description: this.props.envExists ? this.props.environment.description : this.props.repository.description,
+			tag: '',
+			tags: this.props.envExists && Boolean(this.props.environment.tags) ? this.props.environment.tags : [],
+			error: '',
+		}
+    this.getEnvironments = this.getEnvironments.bind(this);
+  };
+
+
+	getEnvironments() {
+		this.props.getEnvironments();
+	};
+
+	handleClose = () => {
```

Comment:
just a note - the syntax in `handleClose` vs `getEnvironments` (& bind in the constructor)produces the same function from a React perspective but there are some subtle differences: https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1

Both are fine just be aware of it.

Reply:
Ah so I have used two different ways to use a class function. First, in the constructor with bind(this) and then outside with an arrow. If I understand correctly, the arrow function will automatically be moved into the constructor which can cause some error during inheritance? So shouldn't the bind method then be the preferred one?